src/hw_tools.py

```
@@ -178,6 +179,89 @@ def install(self, path: Path) -> None:
+                DEBUGDIR={self._storelib_log_dir}
+                #DEBUGLEVEL2MASK
+                #SCSICOMMANDFILTER
+                # Flag to turn Simulation Mode (0-No Simulation, 1- Simulation Mode (Requires simlib.dll))
+                #SIMULATION
+                #LIBPATH
+                #MAXDRVRBUFSIZE
+                #VENDORID
+                """
+            ).strip()
+            + "\n"
+        )
+
+    def _generate_storelib_config(self) -> bool:
+        """Generate configuration file for storelib library logging.
+
+        Workaround to address the issue from
+        [#424](https://github.com/canonical/hardware-observer-operator/issues/424).
+        """
+        # Create the directory for storelib logs
+        self._storelib_log_dir.mkdir(parents=True, exist_ok=True)
+
+        # Check if file exists and log warning before overwriting
+        if self._config_file_path.exists():
+            logger.warning(
+                f"Storelib config file at {self._config_file_path} already exists. Overwriting it."
+            )
+
+        # Write the config file
+        try:
+            with open(self._config_file_path, "w") as f:
+                f.write(self._storelib_config_content)
+            logger.info(f"Created storelib config file at {self._config_file_path}")
+            return True
+        except (IOError, PermissionError) as e:
+            logger.error(f"Failed to write storelib config file: {e}")
+            return False
+
+    def _remove_storelib_config(self) -> None:
+        """Remove the storelib configuration file."""
+        try:
+            if self._config_file_path.exists():
+                self._config_file_path.unlink()
+                logger.info(f"Removed storelib configuration file at {self._config_file_path}")
+            else:
+                logger.info(f"Storelib config file at {self._config_file_path} does not exist")
+        except Exception as e:
```

I think we need to raise the exception